src/main/java/cat/udl/eps/entsoftarch/webingogeiapi/repository/CardRepository.java

```diff
@@ -8,11 +8,15 @@
 import org.springframework.data.rest.core.annotation.RepositoryRestResource;
 
 import java.util.List;
+import java.util.Optional;
+
 @RepositoryRestResource
 
 
 public interface CardRepository extends PagingAndSortingRepository<Card, Integer> {
 
+    Optional<Card> findById(@Param("id") long id);
```

Comment:
PagingAndSortingRepository already implements findById (https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/PagingAndSortingRepository.html)
Moreover, the id defined is Integer, not long. If it needs to be a long instead, better to change it in the definition of the repository interface